### DISCUSSION TITLE
@types/react-table does not recognize Footer on Column

### DISCUSSION THREAD

**Discussion by MonkeyDZeke** [TanStack/table#2882] [Q&A]
I'm trying to use a `Footer` by setting it in the columns array like so:

```
const columns: Column<myDataType>[] = [
  {
    Header: 'Column Name',
    accessor: col => col.name,
    Footer: 'Footer Name',
  },
]
```

But I'm getting a Types error saying that 'Footer' does not exist in type 'Column<myDataType>'. I went and looked in the index.d.ts and searched for 'Footer' and sure enough, it is never mentioned as an accepted property for a Column. Am I doing something wrong, perchance, with how I'm using react-table? If not, does anyone have advice on how to make the types work for me locally until it can get fixed in @types/react-table?

**Comment by timngo127** [comment]
I am having the same issue. Have you found a solution?

**Reply by MonkeyDZeke** [reply]
I have not yet, unfortunately. I went to try and modify the types directly but that ended up a mess.

**Comment by uaite** [comment]
I also have this problem, if anyone has found a solution. I'm using `// @ts-expect-error` against my will for now.

**Comment by timngo127** [comment]
No, I haven't found a solution yet.
    On Tuesday, January 5, 2021, 03:40:26 PM EST, Eduardo Carvalho <notifications@github.com> wrote:  
 
 


I also have this problem, if anyone has found a solution. I'm using // @ts-expect-error against my will for now.

**Comment by baeharam** [comment]
Same problem here...

**Comment by minemanemo** [comment]
I solved it by setting it like this:

- focus is that do not modify `@types/react-table`.

- Follow code is my `react-table-config.d.ts` (this code is base to [URL](https://gist.github.com/ggascoigne/646e14c9d54258e40588a13aabf0102d))

```
import { UseTableColumnOptions as BasicUseTableColumnOptions } from 'react-table'

declare module 'react-table' {

  export type FooterProps<D extends object> = TableInstance<D>

  export interface UseTableColumnOptions<D extends object>
    extends BasicUseTableColumnOptions<D> {
    Footer?: Renderer<FooterProps<D>>
  }
}
```
